torch/csrc/jit/tensorexpr/scripts/bisect.py

```diff
@@ -0,0 +1,68 @@
+import subprocess
+import click
+
+
+def test(cmd, limit):
+    print(f"Testing PYTORCH_JIT_OPT_LIMIT=tensorexpr_fuser={limit} {cmd}")
+    p = subprocess.run(
+        f"PYTORCH_JIT_OPT_LIMIT=tensorexpr_fuser={limit} {cmd}",
+        shell=True,
+        capture_output=True,
+        encoding="utf-8",
+    )
+    print(p.stdout)
+    f = "INTERNAL ASSERT FAILED"
+    if f in p.stdout or f in p.stderr:
+        print("skip")
+        return -1
+    if p.returncode == 0:
+        print("good")
+        return 1
+    print("bad")
+    return 0
+
+
+@click.command()
+@click.option("--cmd")
+def bisect(cmd):
+    last_good = 0
+    first_bad = 10000
+    skips = set()
+
+    # Test if there are any unskipped commits in (last_good, first_bad)
+    def keep_going():
```

Comment:
Checking the `limit` next to `last_good` is sufficient?
```
def keep_going():
  return (last_good+1) not in skips and (last_good+1)!=first_bad
```

Reply:
I'm not sure if this is actually good enough -- I'm not saying it's not, but I found it really hard to be certain that what I was writing was correct, so I went with something that looked pretty obviously correct.  I think the explicit range check is pretty easy to reason about, and even though it's O(n), it's a pretty cheap O(n) since it's just checking set membership.